node/core/dispute-coordinator/src/scraping/mod.rs

```diff
@@ -69,9 +69,56 @@ impl ScrapedUpdates {
 	}
 }
 
+/// A structure meant to facilitate chain reversions in the event of a dispute
+/// concluding against a candidate. Each candidate hash maps to a vector of block
+/// number + hash pairs for all blocks which included the candidate. The entries
+/// in each vector are ordered by decreasing parent block number to facilitate
+/// minimal cost pruning.
+pub struct InclusionsPerCandidate {
+	inclusions_inner: HashMap<CandidateHash, Vec<(BlockNumber, Hash)>>,
```

Comment:
Also pruning becomes more straight forward [that way](https://github.com/paritytech/polkadot/blob/dfa8c5ae321e0523f88fdc43afc6d743812ee49e/node/core/dispute-coordinator/src/db/v1.rs#L340).

Reply:
I think we ran into a similar issue in candidates.rs [here](https://github.com/paritytech/polkadot/blob/master/node/core/dispute-coordinator/src/scraping/candidates.rs#:~:text=Without%20this%20tracking%20we%20won%27t%20be%20able%20to%20remove%20all%20candidates%20before%20block%20X.)

We want to key off of CandidateHash for accessing including block info, but we need to key off of block number for the purpose of removing old entries. 

As is argued in candidates.rs, I don't think this can be done using a single map data structure. If you want, I could use two different BTreeMaps to allow both kinds of access. Alternatively, the single map of sorted vectors does the trick.